projects/plugins/jetpack/extensions/blocks/map/map.php

```diff
@@ -165,3 +165,31 @@ function render_single_block_page() {
 	exit;
 }
 add_action( 'wp', __NAMESPACE__ . '\render_single_block_page' );
+
+/**
+ * Helper function to generate the markup of the block in PHP.
+ *
+ * @param Array $points - Array containing geo location points.
+ *
+ * @return string Markup for the jetpack/map block.
+ */
+function map_block_from_geo_points( $points ) {
+	$map_block_data = array(
+		'points'    => $points,
+		'zoom'      => 1,
+		'mapCenter' => array(
+			'lng' => $points[0]['coordinates']['longitude'],
+			'lat' => $points[0]['coordinates']['latitude'],
+		),
+	);
+
+	$map_block  = '<!-- wp:jetpack/map ' . wp_json_encode( $map_block_data ) . ' -->' . PHP_EOL;
+	$map_block .= sprintf(
+		'<div class="wp-block-jetpack-map" data-map-style="default" data-map-details="true" data-points="%1$s" data-zoom="1" data-map-center="%2$s" data-marker-color="red"></div>',
```

Comment:
Shouldn't we fetch the `zoom` value from the `$map_block_data` array here?

Reply:
> I'm not sure if I'm doing something wrong, but that block appears to be invalid on my end:
> 
> This seems to be because we're missing the fallback content inside the `wp-block-jetpack-map` container. We also appear to miss the `data-show-fullscreen-button="true"` attribute.

The missing data isn't actually necessary to render the block. If you did have an API key saved, then it would be rendered on the front end (right now: a gray square), but the backend would still display that error message.

You're right though - the experience isn't great. I'll see how I can improve it (for both areas).